heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -67,6 +71,10 @@ export default class LintPlugin implements IHeftTaskPlugin<ILintPluginOptions> {
         fix = false;
       }
 
+      const relativeSarifLogPath: string | undefined = pluginOptions?.sarifLogPath;
+      const sarifLogPath: string | undefined =
+        relativeSarifLogPath && path.resolve(heftConfiguration.buildFolderPath, relativeSarifLogPath);
```

Comment:
Technically this can return `string | false` I believe. Use the ternary operator instead:
`relativeSarifLogPath ? path.resolve(...) : undefined`

Reply:
Actually `&&` will preserve the type of the first argument. So `string | undefined` becomes `string` in the truthy code path and `"" | undefined` in the falsy code path.